torch/_inductor/kernel/flex_attention.py

```diff
@@ -855,18 +855,28 @@ def flex_attention(
     for BLOCK_M, BLOCK_N, num_warps, num_stages in configs:
         if SPARSE_KV_BLOCK_SIZE % BLOCK_N != 0 or SPARSE_Q_BLOCK_SIZE % BLOCK_M != 0:
             continue
-        # Work around https://github.com/pytorch/pytorch/issues/129625
-        if num_stages == 2:
-            continue
 
         cur_kernel_options = original_kernel_options.copy()
         # Performance tuning
+        # Triton parameters
+        cur_kernel_options.setdefault("fwd_num_stages", num_stages)
+        cur_kernel_options.setdefault("fwd_num_warps", num_warps)
+        # Work around https://github.com/pytorch/pytorch/issues/129625
+        if cur_kernel_options["fwd_num_stages"] == 2:
+            continue
         cur_kernel_options.setdefault("BLOCK_M", BLOCK_M)
         cur_kernel_options.setdefault("BLOCK_N", BLOCK_N)
         # Blocksparse options
         cur_kernel_options.setdefault("SPARSE_Q_BLOCK_SIZE", SPARSE_Q_BLOCK_SIZE)
         cur_kernel_options.setdefault("SPARSE_KV_BLOCK_SIZE", SPARSE_KV_BLOCK_SIZE)
 
+        for k in list(cur_kernel_options.keys()):
```

Comment:
what is this doing?

Reply:
I might left another comment somwhere can you list all the actual fwd / bwd options we we expect to support

Reply:
fwd kernel options: 
fwd_num_stages, fwd_num_warps, BLOCK_M, BLOCK_N, 
bwd kernel options: 
bwd_num_stages, bwd_num_warps, BLOCK_M1, BLOCK_N1, BLOCK_M2, BLOCK_N2
decoding: 
SPLIT_KV, fwd_num_stages, fwd_num_warps, BLOCK_M, BLOCK_N

shared: 
SPARSE_Q_BLOCK_SIZE, SPARSE_KV_BLOCK_SIZE, SM_SCALE, GQA_SHARED_HEADS, HAS_FULL_BLOCKS, QK_HEAD_DIM, V_HEAD_DIM


things that are currently listed as kernel options but I think they should not be: 
IS_DIVISIBLE, SAFE_M_BOUNDARY, SAFE_N_BOUNDARY(And this is not implemented)

Reply:
This is great accoutning and we should write this down somewhere in code, 

I think I didnt like the string matching for really only 2 keys in each , I know this is forward leaning but at least as of today we dont have an open registration for keys so I am not sure its needed

Reply:
Do you need to pop the bwd args?

Reply:
Hmmm no. It works with bwd args still in it. Just removing prefix for fwd works.